pkg/asset/machines/gcp/machines.go

```diff
@@ -148,3 +154,42 @@ func getNetworks(platform *gcp.Platform, clusterID, role string) (string, string
 		return "", "", fmt.Errorf("unrecognized machine role %s", role)
 	}
 }
+
+// getServiceAccounts retrieves the appropriate service account value based on the credentials mode:
+// Mint: generates a name to match the SA created in Terraform
+// Passthrough: reads the email address from the session credentials
+// Manual: leaves email blank, expecting users to manually edit manifests
+func getServiceAccounts(cm types.CredentialsMode, clusterID, role, projectID string) ([]gcpprovider.GCPServiceAccount, error) {
+	sa := gcpprovider.GCPServiceAccount{
+		Scopes: []string{"https://www.googleapis.com/auth/cloud-platform"},
+	}
+	switch cm {
+	case types.PassthroughCredentialsMode:
+		var err error
+		if sa.Email, err = getServiceAccountEmail(); err != nil {
+			return nil, errors.Wrap(err, "failed to get service account email for passthrough mode")
+		}
+	case types.ManualCredentialsMode:
+		// sa.Email is empty
```

Comment:
We will need to have a separate doc fix to demonstrate how to edit the manifests for manual mode. As this stands, I think install will fail without user intervention. Perhaps I could add something like a platform provision check to make sure the sa.Email value is present.

Reply:
I don't think that we can ask the user to make manual modifications to manifests in order to use a supported install config. If we are going to go the route of requiring the user to supply the service account, then we need to provide that configuration in the install config.

However, both approaches break compatibility with existing install configs, which is not acceptable.